specification/sections/05-events.md

```diff
@@ -137,3 +137,21 @@ See [provider initialization](./02-providers.md#24-initialization) and [setting
 > `PROVIDER_READY` handlers attached after the provider is already in a ready state **MUST** run immediately.
 
 See [provider initialization](./02-providers.md#24-initialization) and [setting a provider](./01-flag-evaluation.md#setting-a-provider).
+
+#### Requirement 5.3.4
+
+> If the provider's `on context changed` executes with mismatched contexts, `PROVIDER_STALE` handlers **MUST** run.
+
+See: [provider events](#51-provider-events), [`provider event types`](../types.md#provider-events)
+
+#### Requirement 5.3.5
+
+> If the provider's `on context changed` function successfully updates the cache, `PROVIDER_READY` handlers **MUST** run.
```

Comment:
Could you explain the use case you're solving with requirement `5.3.5`? I'm worried this could lead to many `PROVIDER_READY` events and overload the meaning behind the event. Would it be helpful to add a new event that could be emitted when the cache updates?

Reply:
I also like the change, and we just had the case in a client project where this would have been helpful.
But I am with you @beeme1mr, maybe it should be another event. Maybe even CONTEXT_CHANGED or so?

Reply:
I like the idea of having another event called `CONTEXT_CHANGED`.

Reply:
Would the provider developer be responsible for emitting the `CONTEXT_CHANGED` event? I think it's worth mentioning that it would only apply to the static context paradigm (i.e. client-side) SDKs.

Reply:
> Would the provider developer be responsible for emitting the `CONTEXT_CHANGED` event? I think it's worth mentioning that it would only apply to the static context paradigm (i.e. client-side) SDKs.

Yep, we'd have to be very clear in the spec this event only applies to to the client.

Reply:
Agreed, this only makes sense in the static paradigm.

@beeme1mr The original problem I was trying to solve with this was thinking about frontend frameworks, mainly React. So the thought is that an integration might want to trigger a refresh to the page If the flag has changed. If you have the page already rendered, and then you login (or set the context in a button callback for whatever reason) we can emit the STALE event to tell any component trying to read the flags to show a spinner until such times as it has finished (using React.Suspense for example). Then when the ready event is fired we can tell those components to load and no longer show a spinner.

I don't think this is an overload of the `PROVIDER_READY` event, because that event's definition is that the provider is ready for flag evaluations, and if the context change has triggered a refetch of the flags and that fetch hasn't completed yet then it isn't ready for evaluations as we know the cached flags are stale.

Reply:
This came up in another context recently... I was helping somebody develop a provider that also goes into a stale state (it downloads a ruleset locally) and when it lost connection with the server, that ruleset could be stale. It was clear the provider should emit `STALE` in this case, but it's a bit unclear what to emit when the provider reconnects and becomes "unstale". This was in a "server-side" context.

If we decide transitioning out of staleness is a valid use of the `READY` event, we should more rigorously specify that in our glossary and in-line docs.

One small _potential_ mismatch here is that provider states include only `NOT_READY` (need to have their `initialize()` called) `READY` (already initialized, or not in need or initialization) or `ERROR` (something went wrong during initialize). See https://openfeature.dev/specification/types#provider-status.

I'm not sure if we should actually add a `STALE` _state_, but we might want to if there's any behavioral difference between a stale/ready provider from a usage perspective. Maybe not though, since I don't think it's relevant to the life-cycle of the provider (init/shutdown).

cc @bacherfl

Reply:
+1 on the STALE _state_ I think the matching of the state and events is a good concept!

Reply:
Should this be included in the current PR?

Reply:
This is my only open question. Should the provider's state be updated to "STALE" while running the `on context changed` method?

Reply:
related: https://github.com/open-feature/spec/pull/200#issuecomment-1675221377